platform-sdk/swirlds-jasperdb/src/main/java/com/swirlds/merkledb/MerkleDbStatistics.java

```diff
@@ -77,43 +90,18 @@ public class MerkleDbStatistics {
     private LongAccumulator flushLeafKeysWritten;
     private DoubleAccumulator flushLeafKeysStoreFileSizeMb;
 
-    /** Hashes store small compactions - time in ms */
-    private LongAccumulator hashesStoreSmallCompactionTimeMs;
-    /** Hashes store small compactions - saved space in Mb */
-    private DoubleAccumulator hashesStoreSmallCompactionSavedSpaceMb;
-    /** Hashes store medium compactions - time in ms */
-    private LongAccumulator hashesStoreMediumCompactionTimeMs;
-    /** Hashes store medium compactions - saved space in Mb */
-    private DoubleAccumulator hashesStoreMediumCompactionSavedSpaceMb;
-    /** Hashes store full compactions - time in ms */
-    private LongAccumulator hashesStoreFullCompactionTimeMs;
-    /** Hashes store full compactions - saved space in Mb */
-    private DoubleAccumulator hashesStoreFullCompactionSavedSpaceMb;
+    /** Hashes store compactions - time in ms */
+    private final List<LongAccumulator> hashesStoreCompactionTimeMsList;
```

Comment:
I propose to change all these lists to (concurrent?) maps, compaction level to accumulators. For every level, an accumulator may be created lazily on the first request to update it. It would let get rid of max compaction level limit.

Reply:
How would you suggest to register the metrics? As of now, all the metrics have to be created in `MerkleDbStatistics#registerMetrics`, and `metrics` instance is not a part of this class. Do you propose to have it in the class?

Reply:
I created a separate ticket to support unlimited number of compaction levels

https://github.com/hashgraph/hedera-services/issues/9291